### ISSUE TITLE
Unexpected `\r` at the end of line

### ISSUE THREAD

**Issue by raynix** [buildkite/feedback#279]
When using something like `export MYLISTS=$(buildkite-agent meta-data get "mylist")` where mylist has multiple lines from the form input, I got a `\r` at the end of each line. Is this expected? I though it should be `\n`.

**Comment by toolmantim** [comment]
@raynix is this on Windows or Linux? We might have a bug for Windows here.

**Comment by raynix** [comment]
@toolmantim sorry I didn't get notified for your reply... We run agents on containers based on Debian stretch docker image.

**Comment by toolmantim** [comment]
@raynix sorry, no worries. Ah, I think I understand the context now… was this a click to unblock w/ fields in the web UI, where they entered multiple lines? Do you know what browser they were using? Was it Windows, Mac or Linux?

**Comment by raynix** [comment]
Yes this is regarding the text field of a block command. I used Chrome on OS X @toolmantim

**Comment by toolmantim** [comment]
Thanks @raynix — that's all we need to investigate.

**Comment by ticky** [comment]
Okay, I’ve figured out why and where this is going on. jQuery is in charge of submitting that form, and it  (non-optionally?) “normalises” form value newlines to Windows-style cr-lf: https://github.com/jquery/jquery/blob/2.1.1/src/serialize.js#L82-L107

**Comment by toolmantim** [comment]
Ah! That'd be it @ticky. Looks like something we might need to fix on the backend then.

**Comment by raynix** [comment]
@ticky nice find! Shame shame jQuery :laughing:

**Comment by ticky** [comment]
@toolmantim I’m not sure there’s a “right” way to fix it on the backend. The “right” way might actually be for the agent to normalise (optionally) to the current platform’s native line endings (`\r\n` on Windows, `\n` on Unix, `\r` on Mac OS 9 and ea– can Go build on Mac OS 9??? 🤔).

Alternately, I did consider moving this page away from jQuery as it’s already constructed within React - then we could at least confidently say “what you put in is exactly what you get out” 🤦🏻‍♀️

**Comment by ticky** [comment]
An alternative which came to mind over lunch is… we could patch our vendored jQuery to not do this. 😂

**Comment by ticky** [comment]
Okay, I believe I have a viable patch - the question is what’s the right move for this. Should we normalise to `\n`, or not normalise at all?

**Comment by toolmantim** [comment]
I’d suggest normalizing to `\n`

**Comment by ticky** [comment]
The trouble with that is it potentially breaks things for Windows users, and I don’t know what the suggestion is for them to work around it?

**Comment by toolmantim** [comment]
I’d say normalize it to `\n` and document it, because at least it’s consistent between the people submitting fields in unblock steps, and you can write a script against the one format?

**Comment by ticky** [comment]
Okay I’ve updated the PRs to do so!

**Comment by raynix** [comment]
Consistency FTW!
